tests/autodiff/matrix-row-major-dedup.slang

```diff
@@ -0,0 +1,22 @@
+//TEST:SIMPLE(filecheck=CHECK): -target hlsl -stage compute -entry computeMain
+
+// This test verifies that row_major and column_major matrices don't create
+// duplicate DiffPair structs when used together in autodiff code.
+// Before the fix, this would generate compilation errors due to mismatched
+// DiffPair_matrixx3Cfloatx2C3x2C3x3E_0 and DiffPair_matrixx3Cfloatx2C3x2C3x3E_1 types.
+
+[Differentiable]
+float3 matmul33(no_diff float3 v, row_major float3x3 w) {
+    return mul(w, v);
+}
+
+[Shader("compute")]
+[NumThreads(1, 1, 1)]
+void computeMain(int3 dispathThreadID: SV_DispatchThreadID) {
+    row_major float3x3 w = float3x3(1.5,2.0,3.1, 5.2,8.8,9.4, 2.6,8.3,2.7);
+    float3 v = float3(0.5, 0.2, 0.3);
+
+    DifferentialPair<row_major float3x3> dpW = diffPair(w);
+    bwd_diff(matmul33)(v, dpW, float3(1));
```

Comment:
This test case is problematic because it didn't write the result of `dpW` into a buffer, so the entire function body is dead code eliminated before even hitting autodiff.

You should modify this test to make sure no dead code removal is happening and test again.

Also you should craft this as an executable test, you should be able to run the test on CPU with `//TEST(compute):COMPARE_COMPUTE: -cpu -output-using-type`